cpp/lib/include/DirectoryEntry.h

```diff
@@ -38,7 +38,12 @@ class DirectoryEntry {
     std::string tag_;
     unsigned field_length_;
     unsigned field_offset_;
+    mutable std::string cache_;
 public:
+    void setCache(const std::string& cache) const { cache_ = cache; }
```

Comment:
Why do we have this cache facility in this class??  It doesn't seem to have anything to do with a directory entry.  (Also the ampersand is on the wrong side.)